refsheet.cls

```diff
@@ -1,57 +1,212 @@
+%
+%   Produce columns with multicol
+%
+\RequirePackage{multicol}[2018/04/20]
+\AfterEndPreamble{%
+  \begin{multicols}{\refsheet@rscols}}
+\AtEndDocument{%
+  \end{multicols}}
+%
+%%%
+
+%%% REDEFINE TITLE
+%
+\renewcommand{\maketitle}{%
+  {%
+   \begin{center}%
+     \Large \@title \\%
+     \vspace{0.1ex}%
+     \small \@author, \@date\end{center}%
+  }%
+}
+%
+%%%
 
-\ProcessOptions
-
-\LoadClass{article}[2013/02/25]
-
-\RequirePackage{fixltx2e}
-\RequirePackageWithOptions{nag}[2013/03/25 othodox,l2tabu,abort]
-
-%% Page layout
-
-\RequirePackage{geometry}[2013/03/25]
-%\usepackage[landscape,margin=0.5in]{geometry}
-\geometry{landscape,margin=0.5in}
-\RequirePackage{multicol}
-
+%%% DOCUMENT DIVISIONS
+%
+%   Package titlesec for setting spacing
+%
+\RequirePackage{titlesec}
+%
+%   Package nopageno so there are no page numbers
+%   (to keep \section from resetting page style).
+%
+\RequirePackage{nopageno} 
+%
+%   No section numbering
+%
+\setcounter{secnumdepth}{0}
+%
+%   No indentation and superfluous spacing
+%
+\setlength{\parindent}{0pt} 
+\titlespacing{\section}{0pt}{0pt}{0pt}
+\titlespacing{\subsection}{0pt}{0pt}{0pt}
+%
+%%%
+
+%%% COMMANDS AND ENVIRONMENTS
+%
+%   Extended definition of lists with enumitem
+%
+\RequirePackage[inline]{enumitem}[2011/08/28]
+%   Calculating widths, e.g. with \widthof
+\RequirePackage{calc} 
+
+% generate dense lists
+\setlist{noitemsep,topsep=0.05ex,}
+\setlist[itemize]{label=\textendash,leftmargin=*}
+\setlist[description]{leftmargin=!}
+% Provide a list-type for truetype descriptions
+% This is retained for backward compatibility
+\newlist{ttdesc}{description}{1}
+\setlist[ttdesc]{font=\ttfamily}
+% Conveniently assigning the longest label as 
+\newenvironment{rslisttt}[1]
+{
+  \begin{description}[labelwidth=\widthof{\texttt{#1}},font=\ttfamily]
+}{
+  \end{description}
+}
 
-%% Title area
+% Provide a list-type for normal descriptions
+\newenvironment{rslist}[1]
+{
+    \begin{description}[labelwidth=\widthof{#1}]
+}{
+    \end{description}
+}
 
-% \RequirePackage{titling} % Unnecessary
-\renewcommand{\maketitle}{%
-  {\begin{center}\Large \@title \\ \vspace{4pt} \small \@author, \@date\end{center}}%
+% Provide a list-type for bold descriptions
+\newenvironment{rslistbf}[1]
+{
+    \begin{description}[labelwidth=\widthof{\textbf{#1}},font=\bfseries]
+}{
+    \end{description}
 }
 
+% Provide a list-type for italic descriptions
+\newenvironment{rslistit}[1]
+{
+    \begin{description}[labelwidth=\widthof{\textit{#1}},font=\itshape]
+}{
+    \end{description}
+}
 
-%% Document Divisions
+% Provide inline description lists
+\newenvironment{rsinline}[1][\ttfamily]
+{
+  \begin{description*}[font={#1}]
+}{
+    \end{description*}
+}
 
-\RequirePackage{titlesec}
-\RequirePackage{nopageno} % To keep \section from resetting page style
+% Provide column list
+\newenvironment{rscolslist}[1][2]
+{
+  \begin{multicols*}{#1}
+    \begin{itemize*}[itemjoin={{\newline}},afterlabel={\nobreakspace}]
```

Comment:
Why the double-brace for `{{\newline}}`? Only the first layer should be effective unless I'm missing some expansion level downstream.